control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -995,8 +996,18 @@ impl CreateVolumeExe for CreateVolume {
         for replica in candidates.candidates() {
             if replicas.len() >= self.replicas as usize {
                 break;
-            } else if replicas.iter().any(|r| r.node == replica.node) {
-                // don't reuse the same node
+            } else if replicas.iter().any(|r| {
+                r.node == replica.node
+                    || spread_label_is_same(r, replica, context).unwrap_or_else(|error| {
```

Comment:
remind me what are we doing here again?
We've got the candidates list in step "1".
And on step "2" we check if the spread is the same? Is that because the list of candidates includes all replicas on a single list, which would not be "spread"?

Reply:
Here the following thing are done
1. Get the candidate list
2. We check if the if the node on which the replica is already placed is again in the candidate list. If yes then we skip the node, to ensure that replica are present on different node. 
3. Along with step 2 we also check if spread is the same.